bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -0,0 +1,776 @@
+public class Clip2BridgeHandler extends BaseBridgeHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Set.of(THING_TYPE_BRIDGE_API2);
+
+    private static final int FAST_SCHEDULE_MILLI_SECONDS = 500;
+    private static final int APPLICATION_KEY_MAX_TRIES = 600; // i.e. 300 seconds, 5 minutes
+    private static final int RECONNECT_DELAY_SECONDS = 10;
+    private static final int RECONNECT_MAX_TRIES = 5;
+
+    private static final ResourceReference DEVICE = new ResourceReference().setType(ResourceType.DEVICE);
+    private static final ResourceReference ROOM = new ResourceReference().setType(ResourceType.ROOM);
+    private static final ResourceReference ZONE = new ResourceReference().setType(ResourceType.ZONE);
+    private static final ResourceReference BRIDGE = new ResourceReference().setType(ResourceType.BRIDGE);
+    private static final ResourceReference BRIDGE_HOME = new ResourceReference().setType(ResourceType.BRIDGE_HOME);
+    private static final ResourceReference SCENE = new ResourceReference().setType(ResourceType.SCENE);
+
+    /**
+     * List of resource references that need to be mass down loaded.
+     * NOTE: the SCENE resources must be mass down loaded first!
+     */
+    private static final List<ResourceReference> MASS_DOWNLOAD_RESOURCE_REFERENCES = List.of(SCENE, DEVICE, ROOM, ZONE);
+
+    private final Logger logger = LoggerFactory.getLogger(Clip2BridgeHandler.class);
+
+    private final HttpClientFactory httpClientFactory;
+    private final ThingRegistry thingRegistry;
+    private final Bundle bundle;
+    private final LocaleProvider localeProvider;
+    private final TranslationProvider translationProvider;
+
+    private @Nullable Clip2Bridge clip2Bridge;
+    private @Nullable ServiceRegistration<?> trustManagerRegistration;
+    private @Nullable Clip2ThingDiscoveryService discoveryService;
+
+    private @Nullable Future<?> checkConnectionTask;
+    private @Nullable Future<?> updateOnlineStateTask;
+    private @Nullable ScheduledFuture<?> scheduledUpdateTask;
+    private Map<Integer, Future<?>> resourcesEventTasks = new ConcurrentHashMap<>();
+
+    private boolean assetsLoaded;
+    private int applKeyRetriesRemaining;
+    private int connectRetriesRemaining;
+
+    public Clip2BridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory, ThingRegistry thingRegistry,
+            LocaleProvider localeProvider, TranslationProvider translationProvider) {
+        super(bridge);
+        this.httpClientFactory = httpClientFactory;
+        this.thingRegistry = thingRegistry;
+        this.bundle = FrameworkUtil.getBundle(getClass());
+        this.localeProvider = localeProvider;
+        this.translationProvider = translationProvider;
+    }
+
+    /**
+     * Cancel the given task.
+     *
+     * @param cancelTask the task to be cancelled (may be null)
+     * @param mayInterrupt allows cancel() to interrupt the thread.
+     * @return always returns null.
```

Comment:
This seems a little obscure, is this just to assign the return value and save one line for three of the usages, e.g.:
```java
checkConnectionTask = cancelTask(checkConnectionTask, true);
```

rather than:
```java
cancelTask(checkConnectionTask, true);
checkConnectionTask = null;
```

?

If yes, I think it would improve readability to explicitly set null by callers rather than this little "trick".

Reply:
Yes. And yes. Done.